### PR TITLE
Cope with broken foreign key references when upgrading

### PR DESCRIPTION
**Pull request by freeekanayaka**
Signed-off-by: Free Ekanayaka <free.ekanayaka@canonical.com>

### CODE REVIEW

File: lxd/db/migration.go [hunk 1]
```diff
@@ -19,6 +19,25 @@ import (
 // clustering, since in those version all data lives in the cluster database
 // (regardless of whether clustering is actually on or off).
 func LoadPreClusteringData(tx *sql.Tx) (*Dump, error) {
+	// Sanitize broken foreign key references that might be around from the
+	// time where we didn't enforce foreign key constraints.
+	_, err := tx.Exec(`
+DELETE FROM containers_config WHERE container_id NOT IN (SELECT id FROM containers);
+DELETE FROM containers_devices WHERE container_id NOT IN (SELECT id FROM containers);
```
Comment (stgraber) [inline lxd/db/migration.go]: Missing containers_devices_config (checking against containers_devices).

File: lxd/db/migration.go [hunk 2]
```diff
@@ -19,6 +19,25 @@ import (
...
+DELETE FROM containers_profiles WHERE container_id NOT IN (SELECT id FROM containers);
+DELETE FROM images_aliases WHERE image_id NOT IN (SELECT id FROM images);
+DELETE FROM images_properties WHERE image_id NOT IN (SELECT id FROM images);
+DELETE FROM images_source WHERE image_id NOT IN (SELECT id FROM images);
+DELETE FROM networks_config WHERE network_id NOT IN (SELECT id FROM networks);
+DELETE FROM profiles_config WHERE profile_id NOT IN (SELECT id FROM profiles);
+DELETE FROM profiles_devices WHERE profile_id NOT IN (SELECT id FROM profiles);
```
Comment (stgraber) [inline lxd/db/migration.go]: Missing profiles_devices_config (checking against profiles_devices).

File: lxd/db/migration.go [hunk 3]
```diff
@@ -19,6 +19,25 @@ import (
...
+DELETE FROM storage_pools_config WHERE storage_pool_id NOT IN (SELECT id FROM storage_pools);
+DELETE FROM storage_volumes WHERE storage_pool_id NOT IN (SELECT id FROM storage_pools);
```
Comment (stgraber) [inline lxd/db/migration.go]: Missing storage_volumes_config (checking against storage_volumes).

File: lxd/db/migration.go (hunk 1 above)
Comment (freeekanayaka) [reply lxd/db/migration.go]: Fixed.

File: lxd/db/migration.go (hunk 2 above)
Comment (freeekanayaka) [reply lxd/db/migration.go]: Fixed.

File: lxd/db/migration.go (hunk 3 above)
Comment (freeekanayaka) [reply lxd/db/migration.go]: Fixed.

**Comment by freeekanayaka** [comment]
I added DELETE FROM containers_profiles WHERE profile_id NOT IN (SELECT id FROM profiles) too for completeness.

**Pull request merged**